rootex/framework/components/visual/cpu_particles_visual_component.cpp

```diff
@@ -109,7 +109,7 @@ bool CPUParticlesVisualComponent::preRender()
 
 void CPUParticlesVisualComponent::render(RenderPass renderPass)
 {
-	ColorMaterial* material = dynamic_cast<ColorMaterial*>(getMaterial());
+	BasicMaterial* material = dynamic_cast<BasicMaterial*>(getMaterial());
```

Comment:
No need for this dynamic cast now

Reply:
Are you sure that this will be the only material class we will be using? The `dynamic_cast` is a safety measure against possible corruption of material files, in case an incorrect type of material(we may add more material types, hopefully) is assigned so the game doesn't crash.

Reply:
Just use BasicMaterial everywhere. No need to futureproof such a hot portion of code